File: backend/internal/app/models/phone.go
package models

import "time"

type Phone struct {
	ID               string    `json:"id" bson:"_id,omitempty"`
	Serial           string    `json:"serial" bson:"serial"`
	Manufacturer     string    `json:"manufacturer" bson:"manufacturer"`
	Model            string    `json:"model" bson:"model"`
	AndroidVersion   string    `json:"androidVersion" bson:"androidVersion"`
	Cpuabi           string    `json:"cpuabi" bson:"cpuabi"`
	MarketName       string    `json:"marketName" bson:"marketName"`
	MarketNameSymbol string    `json:"marketNameSymbol" bson:"marketNameSymbol"`
	CreateTime       time.Time `json:"createTime" bson:"createTime"`
	UpdateTime       time.Time `json:"updateTime" bson:"updateTime"`
}

File: web/src/api/phone.ts
import request from "@/utils/request";

export interface Phone {
  id: string;
  serial: string; 
  manufacturer: string;
  model: string;
  androidVersion: string;
  cpuabi: string;
  marketName: string;
  marketNameSymbol: string;
}

export const addPhone = (phone: Phone) => {
    return request({
        url: '/api/v1/phone/add',
        method: 'post',
        data: phone
    });
}

export const getAllPhoneInfo = () => {
    return request({
        url: '/api/v1/phone/find',
        method: 'get'
    });
}

export const updataPhone = (phone: Phone) => {
    return request({
        url: '/api/v1/phone/update',
        method: 'put',
        data: phone
    });
}

export const deletePhone = (serial: string) => {    
    return request({
        url: '/api/v1/phone/remove'+`?serial=${serial}`,
        method: 'delete',
    });
}